lib/charliecloud.py

```diff
@@ -1447,6 +1460,12 @@ def file_gzip(path, args=[]):
    if (not hasattr(file_gzip, "gzip")):
       if (shutil.which("pigz") is not None):
          file_gzip.gzip = "pigz"
+         # pigz version 2.4.x and lower use --no-time
+         vpigz = cmd_out([file_gzip.gzip, "--version"],
+                         stderr=True).split(' ')[-1][0:3]
+         major, minor = vpigz.split('.')
+         if (major < '2' or (major == '2' and minor < '4')):
+             args.append('--no-time')
```

Comment:
I think parsing version numbers is too complicated, and we have two better options:

1. Use `-m` unconditionally, if it's accepted by all the `pigz` and `gzip` we care about.
2. Zero the relevant four bytes in the file after compression.